Public/Src/Engine/Scheduler/Graph/PipGraphFragmentManager.cs

```diff
@@ -35,51 +40,97 @@ public class PipGraphFragmentManager : IPipGraphFragmentManager
         private readonly ConcurrentBigMap<long, PipId> m_semiStableHashToPipId = new ConcurrentBigMap<long, PipId>();
         private readonly ConcurrentBigMap<long, DirectoryArtifact> m_semiStableHashToDirectory = new ConcurrentBigMap<long, DirectoryArtifact>();
 
+        private readonly TaskFactory m_taskFactory = new TaskFactory(new DedicatedThreadsTaskScheduler(Environment.ProcessorCount, "Add Pip to Graph"));
+
+        private readonly ConcurrentBigMap<AbsolutePath, (PipGraphFragmentSerializer, Task<bool>)> m_taskMap = new ConcurrentBigMap<AbsolutePath, (PipGraphFragmentSerializer, Task<bool>)>();

```

Comment:
very nit: I would prefer to use names here, but this is up to you. And it seems that unnamed tuples are used everywhere anyways.